hashicorp/consul
# 💜 Cc 7187/purple banner for linking existing clusters

### Description
💜 We love purple banners!

This banner is the start of the work allowing users to link their existing Self managed clusters to HCP!
This banner will appear on the services index page:
- If you have not already linked your cluster (to be implemented once we can detect that)
- If you haven't dismissed the alert yet

And will disappear if you:
- Link your cluster
- Dismiss the alert (we save your choice in `localStorage`)

### Testing & Reproduction steps
- Go to the services page (http://localhost:4200/ui/dc1/services)
- See the beautiful Banner
- Press the buttons (only the informational links and dismiss button currently work)
- When you dismiss the alert, check that when you refresh the page, the alert is no longer there 😮 

### Links
- [Jira](https://hashicorp.atlassian.net/browse/CC-7187)
- [Figma](https://www.figma.com/file/WSB8ycZLMEIi41f3iitEFM/Linking-from-Self-managed-to-HCP-Consul-Central?node-id=1913%3A29668&mode=dev)

### Screenshots
![image](https://github.com/hashicorp/consul/assets/4359781/126de447-6038-471d-82ec-22e883c5915e)


### PR Checklist


### Follow ups
Once the ability to fetch linking status in https://hashicorp.atlassian.net/browse/CC-7145 lands, I'll create a fresh PR that the other checks that the `hcp-link-status` service makes

### References
![image](https://github.com/hashicorp/consul/assets/4359781/6e7af137-1dd5-408c-88ed-3d87718e3e4c)
